src/Domain/Donation.php

```diff
@@ -295,10 +302,10 @@ class Donation extends SalesforceWriteProxy
 
     /**
      * @psalm-suppress PossiblyUnusedProperty - used in DB queries
-     * @var ?DateTime   When a queued message that should lead to a Gift Aid claim was sent.
+     * @var ?DateTimeImmutable When a queued message that should lead to a Gift Aid claim was sent.
      */
     #[ORM\Column(type: 'datetime', nullable: true)]
-    protected ?DateTime $tbgGiftAidRequestQueuedAt = null;
+    protected ?DateTimeImmutable $tbgGiftAidRequestQueuedAt = null;
```

Comment:
I see this doesn't need a DB migration because the type in the schema remains `'datetime'`. But I'm wondering if there might be a risk of confusion later from the PHP type diverging?

Reply:
I don't think the PHP type is diverging from anything is it? The DB doesn't have a distinction between DateTime and DateTimeImmutable and doesn't need one since it doesn't have reference types.

Reply:
The DB engine itself doesn't but Doctrine puts a `DC2Type` comment on the field. I don't really know what it uses that for so I'm not sure if we should worry.

Reply:
Right let me see what that's about and if the diff command will generate a migration...

Reply:
I tried running the diff command and I don't think it does anything. Reading https://www.doctrine-project.org/projects/doctrine-dbal/en/3.7/explanation/dc2type-comments.html I think we're best ignoring the DC2Type comments. 
They don't seem to do anything for us. They could be useful if we wanted to define our SQL schema first and then have the PHP classes autogenerated. They'll be removed when we upgrade to DBAL 4.

Reply:
It does nothing in this case because of the `type: 'datetime'` though I think. But fair enough, sounds fine to ignore.

Reply:
My understanding is that these comments do nothing in any case - DBAL 3 just generates them for backward compatibility with DBAL 2. DBAL 4 won't generate them at all. The ORM doesn't use them, it needs all mapping definitions to be in the class file (or xml mapping or whatever).

See discussion here https://github.com/doctrine/migrations/issues/1441